### PR TITLE
Support skipping fields when serializing.

### LINKED ISSUE #207: Add an annotation to indicate a field should be skipped for serialization
Perhaps a custom serialization plugin would be able to do that as well, but without annotation support, it is cumbersome.

### PR DESCRIPTION
**Pull request by davidmorgan**
Fixes #207

---
This change is [Reviewable](https://reviewable.io/reviews/google/built_value.dart/291)

### CODE REVIEW

File: built_value/lib/built_value.dart [hunk 1]
```diff
@@ -95,7 +95,13 @@ class BuiltValueField {
   /// comparing with `operator==`.
   final bool compare;
 
-  const BuiltValueField({this.compare: true});
+  /// Whether the field is serialized. Defaults to `true`.
+  ///
+  /// If a field is not serialized, it must either be `@nullable` or specify a
+  /// default for deserialization to succeed.
```
Comment (zoechi) [inline built_value/lib/built_value.dart]: specify a default value how?

Comment (davidmorgan) [reply built_value/lib/built_value.dart]: Currently I believe this has to be by hand-coding the builder.

Comment (zoechi) [reply built_value/lib/built_value.dart]: ok, thanks for the feedback

**Comment by naiveai** [comment]
It's still not really specified anywhere where to set default values such that it will succeed when a deserializer attempts to come create a class - is there some piece of documentation I'm missing?

**Comment by MichaelM97** [comment]
> It's still not really specified anywhere where to set default values such that it will succeed when a deserializer attempts to come create a class - is there some piece of documentation I'm missing?

Just annotate the field with `@BuiltValueField(serialize: false)` :)

**Pull request merged**